Sources/Authenticator/Models/AuthFactor.swift

```diff
@@ -0,0 +1,108 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+import AWSCognitoAuthPlugin
+import Foundation
+
+/// Represents an authentication factor that can be used during sign-in
+public enum AuthFactor: Equatable, Hashable {
+    /// Password authentication with optional SRP (Secure Remote Password)
+    case password(srp: Bool = true)
+    
+    /// Email-based one-time password authentication
+    case emailOtp
+    
+    /// SMS-based one-time password authentication
+    case smsOtp
+    
+    /// WebAuthn/Passkey authentication
+    case webAuthn
+}
+
+extension AuthFactor: Codable {}
+
+extension AuthFactor {
+    /// Translates AuthFactor to Amplify AuthFactorType
+    func toAuthFactorType() -> AuthFactorType {
+        switch self {
+        case .password(let srp):
+            return srp ? .passwordSRP : .password
+        case .emailOtp:
+            return .emailOTP
+        case .smsOtp:
+            return .smsOTP
+        case .webAuthn:
+            #if os(iOS) || os(macOS) || os(visionOS)
+            if #available(iOS 17.4, macOS 13.5, visionOS 1.0, *) {
+                return .webAuthn
+            } else {
+                // Fallback to password if WebAuthn not available
+                return .passwordSRP
+            }
+            #else
+            // Fallback to password on unsupported platforms
+            return .passwordSRP
+            #endif
+        }
+    }
+    
+    /// Returns true if this auth factor is a password-based factor (with or without SRP)
+    var isPassword: Bool {
+        if case .password = self {
+            return true
+        }
+        return false
+    }
+    
+    /// Display priority for sorting auth factors
+    /// Lower values appear first: WebAuthn (1), SMS (2), Email (3), Password (4)
+    var displayPriority: Int {
+        switch self {
+        case .webAuthn:
+            return 1
+        case .smsOtp:
+            return 2
+        case .emailOtp:
+            return 3
+        case .password:
+            return 4
+        }
+    }
+}
+
+extension Array where Element == AuthFactor {
```

Comment:
can we add a few unit tests for the `Array` extension? mostly to test sorting criteria